src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang

```diff
@@ -29,8 +29,125 @@ module sonic-bgp-allowed-prefix {
             "Initial revision.";
     }
 
+    typedef bgp-allowed-ipv4-prefix {
+       type string {
+         pattern
+            '(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}'
+          +  '([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])'
+          + '/(([0-9])|([1-2][0-9])|(3[0-2]))'
+          + '( (le|ge) (([0-9])|([1-2][0-9])|(3[0-2])))?';
+       }
+       description
+        "The bgp-allowed-ipv4-prefix type represents an IPv4 address prefix
+         in BGP allowed prefix format.
+
+         The string format is an inet:ipv4-prefix (defined in RFC 6991) value
+         appended with an optional suffix.
+         
+         The optional suffix consists of a string 'le' or 'ge' followed by a
+         number which is less than or equal to 32.";
+    }
+
+    typedef bgp-allowed-ipv6-prefix {
+       type string {
+         pattern '((:|[0-9a-fA-F]{0,4}):)([0-9a-fA-F]{0,4}:){0,5}'
+               + '((([0-9a-fA-F]{0,4}:)?(:|[0-9a-fA-F]{0,4}))|'
+               + '(((25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])\.){3}'
+               + '(25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])))'
+               + '(/(([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))'
+               + '( (le|ge) (([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))?';
+         pattern '(([^:]+:){6}(([^:]+:[^:]+)|(.*\..*)))|'
+               + '((([^:]+:)*[^:]+)?::(([^:]+:)*[^:]+)?)'
+               + '(/.+)'
+               + '( (le|ge) (([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))?';
+       }
+       description
+        "The bgp-allowed-ipv6-prefix type represents an IPv6 address prefix
+         in BGP allowed prefix format.
+
+         The string format is an inet:ipv6-prefix (defined in RFC 6991) value
+         appended with an optional suffix.
+         
+         The optional suffix consists of a string 'le' or 'ge' followed by a
+         number which is less than or equal to 128.";
+    }
+
     container sonic-bgp-allowed-prefix {
         container BGP_ALLOWED_PREFIXES {
```

Comment:
Please check sample YANG udpate PR here: https://github.com/sonic-net/sonic-buildimage/pull/19358. 
Need testcase to verify the changes.

Reply:
Hi @wen587, I've updated unittest. Can you please help review?